src/components/ProductCard.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+interface ProductCardProps {
```

Comment:
here, in the components folder, you can have another folder named Products where you will have 2 separate files for the ts types logic and another one for the rest of component. In the ProductsCard.tsx you will have the template and methods for the component and in the productsCard.types.ts you will have all the interfaces/types.